docs: improve README file (#32)
smoya/asyncapi-adoption-metrics: README.md

# asyncapi-adoption-metrics
Metrics library for measuring AsyncAPI tooling adoption.

It aids in comprehending how AsyncAPI tools are used and adopted, facilitating ongoing improvements to our specifications and tools. Currently is being used by @asyncapi/cli as a POC and next step will be making use of it on @asyncapi/studio as well.

Requires NodeJS v18 or greater.

## Metrics collected
We are collecting the following metrics:

- `asyncapi_adoption.action.invoked`:

> With this metric we are tracking the command executed on the CLI as soon as the command is invoked, so it hasn't been completely executed and not finished yet. We just want to know which commands are used, regardless they have failed or succeeded.

Reviewer: ```suggestion
With this metric we are tracking any action started (but not finished) in a tool. For example, a command that got executed on the [CLI](https://github.com/asyncapi/cli/) but didn't finish yet. We just want to know which commands are used, regardless they have failed or succeeded.
```

- `asyncapi_adoption.action.finished`:

> This metric tracks the command executed once it has already finished, carrying the result of the execution and some metadata based on the AsyncAPI document in place.

Reviewer: ```suggestion
With this metric we are trcking the action executed once it has already finished, carrying the result of the execution and some metadata. 
```

## Examples

> Example for `validate` command after being invoked:

Reviewer: ```suggestion
Example for[CLI](https://github.com/asyncapi/cli/) `validate`command after being invoked:
```

```
asyncapi_adoption.action.invoked        COUNTER { action: 'validate' }  1
```

Example of the data collected when the `validate` command has been executed and finished:
```
asyncapi_adoption.action.finished       COUNTER {
  validation_result: 'valid',
  success: true,
  asyncapi_version: '2.6.0',
  asyncapi_servers: 2,
  asyncapi_channels: 4,
  asyncapi_messages: 3,
  asyncapi_operations_send: 3,
  asyncapi_operations_receive: 1,
  asyncapi_schemas: 52,
  action: 'validate'
  }       1
```

## Data storage

> We are making use of [New Relic API](https://docs.newrelic.com/docs/apis/intro-apis/introduction-new-relic-apis/#rest-api) to send the metrics collected to the _New Relic_ servers, where they are stored, and finally visualized on our own dashboards at the AsyncAPI website.

Reviewer: ```suggestion
We are making use of [New Relic API](https://docs.newrelic.com/docs/apis/intro-apis/introduction-new-relic-apis/#rest-api) to send the metrics collected to the _New Relic_ servers, where they are stored, and finally visualized on our own dashboards at the AsyncAPI website (WIP).
```

Reviewer: Worth to add a section about data privacy, mentioning all data is anonymized and sensitive data will never be shared with us.

Author: Agree! Adding that section to the document 👍

```
asyncapi_adoption.action.invoked        COUNTER { action: 'validate' }  1
```

Example of the data collected when the `validate` command has been executed and finished:
```
asyncapi_adoption.action.finished       COUNTER {
  validation_result: 'valid',
  success: true,
  asyncapi_version: '2.6.0',
  asyncapi_servers: 2,
  asyncapi_channels: 4,
  asyncapi_messages: 3,
  asyncapi_operations_send: 3,
  asyncapi_operations_receive: 1,
  asyncapi_schemas: 52,
  action: 'validate'
  }       1
```

## Data storage
We are making use of [New Relic API](https://docs.newrelic.com/docs/apis/intro-apis/introduction-new-relic-apis/#rest-api) to send the metrics collected to the _New Relic_ servers, where they are stored, and finally visualized on our own dashboards at the AsyncAPI website (WIP).

## Data privacy
All the data is anonymized, ensuring no sensitive data reaches our servers.

Even though metrics collection is enabled by default, you can always disable tracking if you wish. To disable tracking, please run the following command:
`asyncapi config analytics --disable`

Once disabled, if you want to enable tracking back again then run:

> `asyncapi config analytics --enable`

Author: @smoya Not sure if all this info would be necessary since it's just related to CLI use case 🤔

Reviewer: I don't thing it should be here. I would say we should add a statement mentioning all tools should enable a mechanism for disabling metrics.

Author: Fair enough 👍 Gonna change it then.
